open-widget-framework/src/widget_form.js

```diff
@@ -0,0 +1,284 @@
+import React, { Component } from 'react'
+import Select from 'react-select'
+
+import { makeOptions } from './utils'
+import { apiPath } from '../es/utils'
+
+
+class EditWidgetForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentWidgetData: null,
+      widgetClass: null,
+      widgetClassConfiguration: null,
+    }
+    this.onSubmit = this.onSubmit.bind(this)
+  }
+
+  componentDidMount() {
+    /**
+     * Fetch data on widget instances in list from fetchRoute
+     */
+    const { errorHandler, fetchData, widgetListId, widgetId } = this.props
+    fetchData(apiPath('widget', widgetListId, widgetId))
+      .then(data => this.setState({
+        currentWidgetData: data.widgetData,
+        widgetClassConfiguration: data.widgetClassConfigurations,
+        widgetClass: Object.keys(data.widgetClassConfigurations)[0]}))
+      .catch(errorHandler)
+  }
+
+  onSubmit(widgetClass, formData) {
+    const { errorHandler, fetchData, onSubmit, widgetListId, widgetId } = this.props
+    let title = formData.title
```

Comment:
Can be `const`. Additionally, instead of mutating `formData` on the next line by deleting title, you can deconstruct it in this way: `const { title, ...configuration } = formData`. I would expect you might see weird things mutating `formData`, so it's both simpler and safer :smiley: 

Same can be done in the similar code in `NewWidgetForm`